### PR TITLE
WebPrintOperationGtk.cpp: check the Cairo features properly

### PR DESCRIPTION
**Pull request by ThomasDevoogdt**
#### 1d0e5ca6393ed22a04de8bb38fd8146b746aaf89
<pre>
WebPrintOperationGtk.cpp: check the Cairo features properly

<a href="https://bugs.webkit.org/show_bug.cgi?id=279905">https://bugs.webkit.org/show_bug.cgi?id=279905</a>

Reviewed by Michael Catanzaro.

Cairo might be compiled without e.g. pdf, ps or svg support,
just check that to avoid compile errors at build time.

fatal error: cairo-pdf.h: No such file or directory
   43 | #include &lt;cairo-pdf.h&gt;
      |          ^~~~~~~~~~~~~

Canonical link: <a href="https://commits.webkit.org/283866@main">https://commits.webkit.org/283866@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/93e33d824539ae8a2e500bd0b81ce8c690e07602

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/67623 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/47005 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/20262 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/71677 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [⏳ 🛠 wincairo ](https://ews-build.webkit.org/#/builders/Win-Build-EWS "Waiting in queue, processing has not started yet") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/54804 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/18567 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/71677 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [⏳ 🧪 wincairo-tests ](https://ews-build.webkit.org/#/builders/Win-Build-EWS "Waiting in queue, processing has not started yet") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/70690 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/43135 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/58484 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/71677 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/39808 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/17122 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/61768 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/16231 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/73375 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/11586 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/15544 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/61598 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/11621 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/58552 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/61651 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14994 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/9462 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/3076 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/42808 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/43886 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/45075 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/43627 "Built successfully") | | |

### CODE REVIEW

File: Source/WebKit/WebProcess/WebPage/gtk/WebPrintOperationGtk.cpp [hunk 1]
```diff
@@ -278,6 +291,7 @@ void WebPrintOperationGtk::startPage(cairo_t* cr)
     cairo_surface_t* surface = cairo_get_target(cr);
     cairo_surface_type_t surfaceType = cairo_surface_get_type(surface);
     if (surfaceType == CAIRO_SURFACE_TYPE_PS) {
```
Comment (mcatanzaro) [inline Source/WebKit/WebProcess/WebPage/gtk/WebPrintOperationGtk.cpp:293]: The `CAIRO_SURFACE_TYPE` values are defined unconditionally, right?

**Review by mcatanzaro — APPROVED** [review]

File: Source/WebKit/WebProcess/WebPage/gtk/WebPrintOperationGtk.cpp (hunk 1 above)
Comment (ThomasDevoogdt) [reply Source/WebKit/WebProcess/WebPage/gtk/WebPrintOperationGtk.cpp:293]: See https://gitlab.freedesktop.org/cairo/cairo/-/blob/master/src/cairo.h?ref_type=heads#L2565, this is defined in cairo.h.

**Comment by philn** [comment]
Can you remove the signed-off-by please?

**Comment by ThomasDevoogdt** [comment]
> Can you remove the signed-off-by please?

I can do that.

**Comment by aperezdc** [comment]
Backported into the 2.46 branch as commit 95c0fa720efc12edf7d1c159ac4676bbc47d81eb

**Comment by aperezdc** [comment]
Backported into the 2.44 branch as commit 927392a88f29a4d63f1be0ad83f817771a91f6a9

**Comment by ThomasDevoogdt** [comment]
> Backported into the 2.44 branch as commit [927392a](https://github.com/WebKit/WebKit/commit/927392a88f29a4d63f1be0ad83f817771a91f6a9)

Hi, you've forgot to include the global cairo include. See my [remark](https://github.com/WebKit/WebKit/commit/927392a88f29a4d63f1be0ad83f817771a91f6a9#r146948852).

**Comment by aperezdc** [comment]
> > Backported into the 2.44 branch as commit [927392a](https://github.com/WebKit/WebKit/commit/927392a88f29a4d63f1be0ad83f817771a91f6a9)
> 
> Hi, you've forgot to include the global cairo include. See my [remark](https://github.com/WebKit/WebKit/commit/927392a88f29a4d63f1be0ad83f817771a91f6a9#r146948852).

Thanks a ton for noticing, I have pushed 38bc07ea6987437afaa97b452539253cf7c2afdb to the branch with the fix 🙏🏼

**Pull request merged**